File: frontend/src/types/task.ts
export type TaskRunStatus = 'running' | 'success' | 'error' | 'missed'

export interface TaskRunRecord {
  run_id: string
  job_id: string
  job_name: string
  status: TaskRunStatus
  scheduled_at: string | null
  started_at: string | null
  finished_at: string | null
  duration_seconds: number | null
  message: string | null
}

export interface ScheduledTaskStatus {
  job_id: string
  name: string
  exists: boolean
  scheduler_running: boolean
  trigger: string | null
  next_run_time: string | null
  last_run: TaskRunRecord | null
  recent_runs: TaskRunRecord[]
}

File: backend/app/schemas/task.py
from datetime import datetime

from pydantic import BaseModel


class TaskRunRecordResponse(BaseModel):
    run_id: str
    job_id: str
    job_name: str
    status: str
    scheduled_at: datetime | None
    started_at: datetime | None
    finished_at: datetime | None
    duration_seconds: float | None
    message: str | None = None


class ScheduledTaskStatusResponse(BaseModel):
    job_id: str
    name: str
    exists: bool
    scheduler_running: bool
    trigger: str | None
    next_run_time: datetime | None
    last_run: TaskRunRecordResponse | None
    recent_runs: list[TaskRunRecordResponse]
